### PR TITLE
etcd: upgrade go version from 1.21.3 to 1.21.4

### PR DESCRIPTION
**Pull request by arjunmalhotra1**
To keep etcd projects up to date with the latest patch releases & incorporate the latest security updates.

### CODE REVIEW

**Comment by serathius** [comment]
Please sign DCO

**Comment by arjunmalhotra1** [comment]
Thanks @serathius. I have fixed the DCO.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module go.etcd.io/raft/v3
 
-go 1.21
+go 1.21.4
```
Comment (serathius) [inline go.mod:3]: Please don't specify patch version in go module.

File: tools/mod/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module go.etcd.io/raft/tools/v3
 
-go 1.21
+go 1.21.4
```
Comment (serathius) [inline tools/mod/go.mod:3]: ditto

**Comment by arjunmalhotra1** [comment]
Thanks @serathius I have addressed the issue.

**Review by jmhbnz — APPROVED** [review]
LGTM on green - Thanks @arjunmalhotra1

**Review by ahrtr — APPROVED** [review]
lgtm

**Pull request merged**